spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/check_service_retired_spec.rb

```diff
@@ -0,0 +1,55 @@
+require_domain_file
+
+describe ManageIQ::Automate::Service::Retirement::StateMachines::Methods::CheckServiceRetired do
+  let(:admin) { FactoryGirl.create(:user_admin) }
+  let(:request) { FactoryGirl.create(:service_retire_request, :requester => admin) }
+  let(:service) { FactoryGirl.create(:service) }
+  let(:task) { FactoryGirl.create(:service_retire_task, :destination => service, :miq_request => request) }
+  let(:svc_task) { MiqAeMethodService::MiqAeServiceServiceRetireTask.find(task.id) }
+  let(:svc_service) { MiqAeMethodService::MiqAeServiceService.find(service.id) }
+  let(:vm) { FactoryGirl.create(:vm) }
+  let(:retired_vm) { FactoryGirl.create(:vm, :retired => true) }
+  let(:root_object) do
+    Spec::Support::MiqAeMockObject.new('service'             => svc_service,
+                                       'service_retire_task' => task,
+                                       'service_action'      => 'Retirement')
+  end
+  let(:ae_service) { Spec::Support::MiqAeMockService.new(root_object) }
+
+  context "with non retired resource" do
+    it "check" do
+      service.service_resources << FactoryGirl.create(:service_resource, :resource_type => "VmOrTemplate", :service_id => service.id, :resource_id => vm.id)
+      service.save
+      allow(svc_service).to receive(:check_retired)
```

Comment:
@d-m-u 
Should the check_retired be returning a true/false and exhibiting different behaviour based on some earlier setting.